common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -196,7 +195,13 @@ def get_unlink_form_name(self):
 
 def get(request):
     """Gets the running pipeline from the passed request."""
-    return request.session.get('partial_pipeline')
+    strategy = social_django.utils.load_strategy(request)
+    token = strategy.session_get('partial_pipeline_token')
+    partial_object = strategy.partial_load(token)
+    pipeline_data = None
+    if partial_object:
+        pipeline_data = {'kwargs': partial_object.kwargs, 'backend': partial_object.backend}
+    return pipeline_data
```

Comment:
Option B is better; I think the less code changes at this stage that could break things somewhere, the better.